commands/microservice

```diff
@@ -74,35 +166,87 @@ while [[ $# -gt 0 ]]; do
         exit 1
       fi
       ;;
+    -t|--tenant|-tenant)
+      if [[ $# -gt 1 ]]; then
+        tenant="$2"
+        shift
+      else
+        echo "Option $1 requires an argument." >&2
+        exit 1
+      fi
+      ;;
+    --ignoreAlreadyDone|-ignoreAlreadyDone)
+      ignoreAlreadyDone=true
+      ;;
     -h|--help)
       help
       exit 0
       ;;
+    --*)  # Capture options starting with "--"
+      REST_ARGS+=("$1")
+      ;;
     *)
-      echo "Invalid option or argument: $1" >&2
-      help
-      exit 1
       ;;
   esac
   shift
 done
+set -- "${REST_ARGS[@]}" # Get all remain arguments which is not listed in the cases above
+
+# Check id availability
+if [[ -z "$id" ]]; then
+  echo "Option --id is required for the '$command' command." >&2
+  exit 1
+fi
 
 # Execute the command
 case "$command" in
   subscribe)
-    if [[ -z "$id" ]]; then
-      echo "Option --id is required for the 'subscribe' command." >&2
+    info=($(get_microservice_info "$id" "$@"))
+    name="${info[0]}"
+    type="${info[1]}"
+
+    if [[ $type == "ERROR" ]]; then
+      exit 1
+    fi
+
+    if [[ -z "$tenant" ]]; then
+      echo "Subscribe $type $name to current tenant $C8Y_TENANT"
+      return=$(subscribe "$id" "$@")
+    else
+      echo "Subscribe $type $name to tenant $tenant"
+      return=$(subscribe_with_tenant_id "$id" "$tenant" "$@")
+    fi
+
+    if [[ $return -eq 0 ]]; then
+      echo "Subscribed successfully." >&2
```

Comment:
- maybe we could provide a different message if the microservice was already subscribed and the "ignoreAlreadyDone                   " was used?